app/components/open_project/common/main_menu_toggler_component.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+# -- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See COPYRIGHT and LICENSE files for more details.
+# ++
+#
+module OpenProject
+  module Common
+    class MainMenuTogglerComponent < ApplicationComponent
+      def initialize(opened:)
+        super
+
+        @opened = opened
+      end
+
+      def icon
+        @opened ? "sidebar-expand" : :"sidebar-collapse"
+      end
+
+      def id
+        "menu-toggler--#{@opened ? 'opener' : 'closer'}"
+      end
+
+      def aria_label
+        @opened ? I18n.t("js.label_hide_project_menu") : I18n.t("js.label_expand_project_menu")
+      end
+
+      def scheme
+        :invisible
+      end
+
+      def size
+        @opened ? :medium : :small
+      end
+
+      def data
+        {
+          action: "click->menus--main-toggler#toggleNavigation"
+        }
+      end
+
+      def call
+        render(Primer::Beta::IconButton.new(icon:,
+                                            id:,
+                                            aria: { label: aria_label },
+                                            scheme:,
+                                            size:,
+                                            data:))
```

Comment:
It might be worth discussing tooltip direction with the UI/UX team.

```suggestion
                                            data:,
                                            tooltip_direction: @opened ? :se : :sw))
```

For the attribute help texts we went with a SW direction (see #19119 / [OP 64531](https://community.openproject.org/wp/64531). In this case, specifying SW wouldn't change anything in the open/expanded state:

<img width="163" alt="Screenshot 2025-06-18 at 11 07 21" src="https://github.com/user-attachments/assets/d9842347-930d-41a0-98c0-fab2debd66cb" />

But I think SE would be an improvement in the closed/collapsed state:

| Default | South East |
|--------|--------|
| <img width="231" alt="Screenshot 2025-06-18 at 11 07 10" src="https://github.com/user-attachments/assets/279ab31c-eb35-4797-9951-028b12f29705" />  | <img width="228" alt="Screenshot 2025-06-18 at 11 11 17" src="https://github.com/user-attachments/assets/c9a78f10-717f-4740-a45f-77e176c02f07" /> |